src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +36,72 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * @experimental in version 3.3
+     */
+    public function registerClasses(Definition $prototype, $namespace, $directoryGlob)
+    {
+        if ('\\' !== substr($namespace, -1)) {
+            throw new InvalidArgumentException(sprintf('Namespace prefix must end with a "\\": %s.', $namespace));
+        }
+        if (!preg_match('/^(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+\\\\)++$/', $namespace)) {
+            throw new InvalidArgumentException(sprintf('Namespace is not a valid PSR-4 prefix: %s.', $namespace));
+        }
+
+        if (strlen($directoryGlob) === $i = strcspn($directoryGlob, '*?{[')) {
+            $directoryPrefix = $directoryGlob;
+            $directoryGlob = '';
+        } else {
+            $directoryPrefix = dirname(substr($directoryGlob, 0, 1 + $i));
+            $directoryGlob = substr($directoryGlob, strlen($directoryPrefix));
+        }
+
+        $directoryPrefix = $this->locator->locate($directoryPrefix, $this->currentDir, true);
+        $directoryPrefix = realpath($directoryPrefix) ?: $directoryPrefix;
+        $directoryGlob = $directoryPrefix.$directoryGlob;
+
+        // track directories only for new & removed files
+        $this->container->fileExists($directoryPrefix, '/^$/');
+
+        $classes = $this->findClasses($namespace, $directoryGlob, strlen($directoryPrefix));
+
+        foreach ($classes as $class) {
+            $this->container->setDefinition($class, clone $prototype);
+        }
+    }
+
+    private function findClasses($namespace, $directoryGlob, $prefixLen)
+    {
+        $classes = array();
+        $extRegexp = defined('HHVM_VERSION') ? '/\\.(?:php|hh)$/' : '/\\.php$/';
+
+        foreach (glob($directoryGlob, defined('GLOB_BRACE') ? GLOB_BRACE : 0) as $path) {
+            if (is_dir($path)) {
+                $flags = \FilesystemIterator::SKIP_DOTS | \FilesystemIterator::UNIX_PATHS | \FilesystemIterator::FOLLOW_SYMLINKS;
+                $files = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($path, $flags));
+            } else {
+                $files = array($path => new \SplFileInfo($path));
+            }
+
+            foreach ($files as $path => $info) {
+                if (!preg_match($extRegexp, $path, $m) || !$info->isFile() || !$info->isReadable()) {
+                    continue;
+                }
+                $class = $namespace.ltrim(str_replace('/', '\\', substr($path, $prefixLen, -strlen($m[0]))), '\\');
+
+                if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)*+$/', $class)) {
+                    continue;
+                }
+                if (!$r = $this->container->getReflectionClass($class, true)) {
+                    continue;
+                }
+                if (!$r->isInterface() && !$r->isTrait()) {
```

Comment:
nope: as explained in the description, getter injection works also with abstract classes, and that's desired (prevents writting a method with gargabe in it, and explicitly tells consumers that this needs to be overriden)